File: apps/cargo-tracker/src/main/java/com/example/cargotracker/shipper/domain/event/ShipperRegisteredEvent.java
package com.example.cargotracker.shipper.domain.event;

import com.example.cargotracker.shipper.domain.model.CustomerCategory;
import com.example.cargotracker.shipper.domain.model.ShipperId;

public record ShipperRegisteredEvent(ShipperId shipperId, CustomerCategory category) {
}

File: docs/article/source/csharp/apps/cargo-tracker/src/CargoTracker.Web/Shipper/Domain/Model/Events/ShipperRegisteredEvent.cs
using CargoTracker.Shared.Domain.Model;
using MediatR;

namespace CargoTracker.Shipper.Domain.Model.Events;

/// <summary>荷主が新規登録されたことを表すドメインイベント。</summary>
public sealed record ShipperRegisteredEvent(ShipperId ShipperId, ShipperType Type) : INotification;
